lib/features/auth/domain/usecases/login.dart

```diff
@@ -0,0 +1,31 @@
+import 'package:dartz/dartz.dart';
+import 'package:equatable/equatable.dart';
+import 'package:final_project/core/domain/error/failure.dart';
+import 'package:final_project/core/domain/usecases/usecase.dart';
+import 'package:final_project/features/auth/domain/entities/login_entity.dart';
+import 'package:final_project/features/auth/domain/repositories/auth_repository.dart';
+import 'package:injectable/injectable.dart';
+
+@injectable
+class Login implements UseCase<Unit, LoginData> {
```

Comment:
Stick to a naming convention for UseCases like LoginUseCase and login_use_case.dart for files. This helps alot in searching for files within the IDE/Editor.